### DISCUSSION TITLE
i'm thinking stop open-source for branch v2

### DISCUSSION THREAD

**Discussion by rurirei** [rurirei/Kitsunebi#19] [Ideas]
i'm thinking stop open-source for branch v2.

i thought open source may raises more contribution and community development, this takes a mistake sadly.

i will make the new app, release without source code, even this may violate GPL license of some.

i will make most of code open source (this repo based), without some code my own.

say your opinion here.

**Comment by ghost** [comment]
thanks a lot,it is hard to continue

**Comment by eroxs** [comment]
@rurirei  Can you give me an email or a telegram id?
I want a programmer to run xray core on Android and I will pay for this collaboration.

**Reply by rurirei** [reply]
github discussion please.

**Comment by AkinoKaede** [comment]
Hmm. I think open source is important against anti-censorship software.

**Reply by rurirei** [reply]
v2ray core will open source at my repo (rurirei/v2ray-core/rkt).

**Reply by eroxs** [reply]
what about androidlibv2rayLite ?

**Reply by rurirei** [reply]
> what about androidlibv2rayLite ?

@eroxs it is not of mine.

**Comment by rurirei** [comment]
i'm thinking make a open source without some features, to looking for possible contribution, and for this i will do no release for this open-source branch. what is it?

@AkinoKaede opinion welcome.

**Reply by AkinoKaede** [reply]
I don't think that the purpose of open source is to obtain the contributions of others, but to protect the freedom of users. So I hope that users can access all the source code, which is particularly important for anti-censorship software.

**Reply by rurirei** [reply]
thanks for your opinion. but i make this open source for contribution, i do not make a service for public.

**Reply by AkinoKaede** [reply]
Thanks. BTW, you can change the license if you got all contributors' permission.

**Reply by rurirei** [reply]
😿 this repo has no contributors sadly.

**Comment by rurirei** [comment]
addition, i will not violate GPL license of some if i do not make releases.

**Reply by rurirei** [reply]
@AkinoKaede as so.

**Reply by AkinoKaede** [reply]
Of course, the GPL only requires you to make the code available if you release the software to public.

**Comment by ercxar** [comment]
As someone who doesn't have the knowledge to contribute, the option to use such apps are very limited on each platform. Most of them are out of date and only a few devs are maintaining their apps.
If you have plans to make a new app and probably sell it, it's all fine by me as long as you continue maintaining your app with the most recent updates on all aspects. I'm happy to pay the price for your efforts.

**Comment by thebadgateway** [comment]
I am willing to donate to keep the code open

**Comment by rurirei** [comment]
i may make partly open source for new app [rurirei/Clover](https://github.com/rurirei/Clover), no app release as you could build it.

**Comment by littlef0x** [comment]
> 
> 
> Hmm. I think open source is important against anti-censorship software.

I agree.

> 
> 
> i'm thinking make a open source without some features, to looking for possible contribution, and for this i will do no release for this open-source branch. what is it?

Partial open source is a good choice. But I do not hope the open source branch to be laking in feature. Users who care more about open source are more likely to contribute to the app and the comminuty. It is unfair to exclude them from these features. I think it would be better if you can keep the core functions open source, from which most security concerns originate, and the UI part close source.

BTW. It's OK to sell it on Google Play as long as it does not violate the license. And I will pay for it.

**Reply by rurirei** [reply]
i'm preferred partly open source.

**Reply by littlef0x** [reply]
I agree. But the point is in which way: separating some features or separating the interface. I prefer the latter one. However, I totally understand if the author prefers the other way due to either technical or personal consideration. It is hard to develop anti-censorship tools nowadays. Whatever you choose, I'd like to express my gratitude here.

**Reply by rurirei** [reply]
i'm preferred in the former, to open source the android app most and core interface less.

**Comment by rurirei** [comment]
😿 
![Screenshot_20210418-202850_1](https://user-images.githubusercontent.com/72071920/115145610-0599eb80-a085-11eb-8f67-aba9ed0043db.jpg)

**Reply by littlef0x** [reply]
Well, I think this meme actually refers to open source software developed by individual person vs. commercial software by large company, since open source softwares were often developed by individual person, while large companies seldom wrote open source softwares. Things have changed these years, so it is inappropriate now. ╮(╯▽╰)╭

**Comment by rurirei** [comment]
i'm much excited to worked [those features](https://github.com/rurirei/Clover/discussions/1) out. may make a release for preview in latter.
